src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1183,14 +1183,18 @@ private void ParseModifiers(SyntaxListBuilder tokens)
                 {
                     case DeclarationModifiers.Partial:
                         var nextToken = PeekToken(1);
-                        var isPartialType = this.IsPartialType();
-                        var isPartialMember = this.IsPartialMember();
-                        if (isPartialType || isPartialMember)
+                        var isPartialFollowedByType = this.IsPartialType();
+                        var isPartialType = isPartialFollowedByType || IsNonContextualModifier(nextToken);
```

Comment:
What's the result if we try to parse  `partial partial ref struct`?

Reply:
I think the diagnostic will be similar to we have today (expected `;` etc). I'll add the test.